clients/java/signalr/src/test/java/com/microsoft/aspnet/signalr/HubConnectionTest.java

```diff
@@ -841,24 +842,43 @@ public void callingStartOnStartedHubConnectionNoOps() throws Exception {
     @Test
     public void cannotSendBeforeStart() throws Exception {
         Transport mockTransport = new MockTransport();
-        HubConnection hubConnection = new HubConnection("http://example.com", mockTransport);
+        HubConnection hubConnection = new HubConnection("http://example.com", mockTransport, true);
         assertEquals(HubConnectionState.DISCONNECTED, hubConnection.getConnectionState());
 
         Throwable exception = assertThrows(HubException.class, () -> hubConnection.send("inc"));
         assertEquals("The 'send' method cannot be called if the connection is not active", exception.getMessage());
     }
 
+    @Test
+    public void negotiateSentOnStart() throws Exception {
```

Comment:
This is a good start. We're also going to want to add a test that verifies the redirect behavior.